core/src/main/java/io/aiven/kafka/tieredstorage/metrics/CaffeineStatsCounter.java

```diff
@@ -135,11 +155,13 @@ private void initSensor(final String sensorName, final String metricName, final
     @Override
     public void recordHits(final int count) {
         cacheHitCount.add(count);
+        updateCacheSize();
     }
 
     @Override
     public void recordMisses(final int count) {
         cacheMissCount.add(count);
+        incrementCacheSize(); // increment by one when missing
```

Comment:
Could you elaborate why we are incrementing the size here?

Reply:
Sure. Initially, I started by only adding the size supplier on `recordHit`; but metrics will stay unmodified until there's a hit, which is inconvenient.
To solve this I added this hook to increment by one, as a "miss" should mean that the cache will increase.

Another alternative may be to call the size supplier here but haven't tested if the value is already updated or not.

Reply:
I trying to say a different thing, we are ourselves calling `recordMiss` and there we are not doing anything to size metric. At the same time this method is called by Caffeine and I'm not sure if we need to do anything to size here at all of if we even get here for ChunkCache at least since we are calling `compute` all the time.